src/main/java/net/snowflake/client/core/auth/wif/OidcIdentityAttestationCreator.java

```diff
@@ -12,8 +10,24 @@ public class OidcIdentityAttestationCreator implements WorkloadIdentityAttestati
   private static final SFLogger logger =
       SFLoggerFactory.getLogger(OidcIdentityAttestationCreator.class);
 
+  private final String token;
+
+  public OidcIdentityAttestationCreator(String token) {
+    this.token = token;
+  }
+
   @Override
-  public WorkloadIdentityAttestation createAttestation() throws SFException {
-    throw new SFException(ErrorCode.FEATURE_UNSUPPORTED, "OIDC Workload Identity not supported");
+  public WorkloadIdentityAttestation createAttestation() {
+    if (token == null) {
+      logger.debug("No OIDC token was specified");
+      return null;
+    }
+    WorkloadIdentityUtil.JwtClaims claims = WorkloadIdentityUtil.extractAndVerifyClaims(token);
+    if (claims == null) {
+      logger.error("Could not extract and verify claims from token");
+      return null;
+    }
+    return new WorkloadIdentityAttestation(
+        WorkloadIdentityProviderType.OIDC, token, claims.toMap());
```

Comment:
nit: suggest explicitly limiting this to sub and aud, unless we rename the `JwtClaims` class to `SubjectAndIssuer`. We don't want to include other claims in here, otherwise we'll never be able to cache this identity.

```suggestion
        WorkloadIdentityProviderType.OIDC, token, Map.of("sub", claims.getSubject(), "iss", claims.getIssuer()));```

Reply:
Wasn't sure about how extensible would that be. Renamed.